framework/src/st_things/things_stack/src/common/framework/things_data_manager.c

```diff
@@ -872,6 +872,9 @@ static int parse_things_info_json(const char *filename)
 					cJSON *vendor_id = cJSON_GetObjectItem(spec_platform, KEY_DEVICE_SPECIFICATION_PLATFORM_VENDORID);
 
 					if (NULL != manufacturer_name) {
+						if(strlen(manufacturer_name->valuestring) != 4) {
+							return 1;
```

Comment:
i think you shoud "return 0"

Reply:
also, please add a log the reason for return.

Reply:
modify